test/Prop.hs

```diff
@@ -990,6 +993,56 @@ monadBind constr eq t (a, b) = withMaxSuccess maxTestCount $
         let list = a >>= \x -> (+ x) <$> b
         listEquals eq stream list
 
+-------------------------------------------------------------------------------
+-- Tests for S.groupsBy
+-------------------------------------------------------------------------------
+
+-- |
+-- If the list is empty, returns Nothing,
+-- else wraps the minimum value of the list in Just.
+maybeMinimum :: [Int] -> Maybe Int
+maybeMinimum [] = Nothing
+maybeMinimum ls = Just $ minimum ls
+
+
+-- |
```

Comment:
See https://github.com/composewell/haskell-dev/blob/master/coding-style.rst#haddock for how to write the haddock comments. We do not use `-- |`` on a line of its own.

Reply:
oh but you merged this ?

Reply:
That does not stop the merge but something that you should keep in mind in the next commit.